src/components/Card/index.tsx

```diff
@@ -1,3 +1,165 @@
-export const Card = () => {
-  return <></>;
+'use client';
+import * as S from './style';
+import * as I from 'assets/imgs';
+import * as C from 'components';
+import Image from 'next/image';
+import projectData from 'interface/projectData';
+import React, { useState } from 'react';
+import { Detail, Vector } from 'assets/imgs';
+import Link from 'next/link';
+const Card = ({ data }: { data: any }) => {
```

Comment:
type 지정을 해준다면 좋을것 같습니다!

Reply:
```suggestion
const Card = ({ projectname, creater }: { projectname:string; creater:string; })
```

혹시 이런 형태의 `type` 지정을 말씀하시는건가요 ?

Reply:
음.. 생각 해보니 MainContainer에서 Card들이 들어갈 틀을 잡아준 후 map을 돌려 Card 컴포넌트를 뿌려주는게 각 컴포넌트의 목적에 적합하지 않나 싶습니다
const [projectList, setProjectList] = useState<projectData[]>(data);에서 setProjectList가 사용되어지지 않은 것으로 보이는데
굳이 state 선언할 필요도 없구요
타입 선언도 Card 컴포넌트에서 import projectData from 'interface/projectData'; 가져다써서 깔끔하게 되지 않을까 생각합니다

전예빈씨 생각은 어떤가요?


```suggestion
import project from '../../../public/data/project.json';

const MainContainer = ({ isDark }: Props) => {
  return (
    <S.MainBox>
      <C.Banner />
      <S.CardContainer>
        <S.Cards>
            <S.SlideBox
              style={{
                display: 'flex',
                position: 'absolute',
                left: `${-slideIndex * 100}%`, // 슬라이드 이동에 따라 left 값을 변경
                top: '0',
                transition: 'left 0.3s ease-in-out',
              }}
            >
            {
                project.map((data, i) => (
                    <C.Card data={data} key={i} />
                )
            }
            </S.SlideBox>
        </S.Cards>
      <Image
        src={Vector}
        alt=""
        style={{
          transform: 'matrix(-1, 0, 0, 1, 0, 0)',
          marginLeft: '3.125rem',
          zIndex: '1000000000',
          cursor: 'pointer',
        }}
        onClick={handleNextSlide}
      />
    </S.Container>
      </S.CardContainer>
    </S.MainBox>
  );
};



// Card
import projectData from 'interface/projectData';

const Card = ({ data }:projectData ) => {}

```

Reply:
일단 state 는 무조건 선언을 해줘야 합니다 !
왜냐하면 json은 일반적인 데이터일 뿐이지만 typescript 에서는 타입정보를 제공해줘야 하거든용
그래서 일반적으로 `타입 캐스팅`이나 `인터페이스`를 사용하는거구요
그리고 map 함수를 사용하려면 json 을 배열로 변환해줘야 하는데 타입지정과 배열변환을 한꺼번에 해주는 코드가
 `const [projectList, setProjectList] = useState<projectData[]>(data);`
 이 친구 이구용

Reply:
그 외에 카드는 MainContainer 에서 하는게 맞을 것 같네요

Reply:
IM SORRY